cppcache/integration/test/PartitionRegionOpsTest.cpp

```diff
@@ -46,11 +48,21 @@ using apache::geode::client::RegionShortcut;
 
 using std::chrono::minutes;
 
+std::string getClientLogName() {
```

Comment:
Oof... Man, I looked into why you're doing this, and I get you, and I'm just as sorry.

Can I convince you to add a refactor that will eliminate all this dubious file handling? If you look at cppcache/src/Log.cpp, there are 2 places, 2 lines each, where we fprintf(stdout... and a flush. What if you replaced both those with a write to ::std::cerr? No need to flush, cerr has unitbuf enabled by default.

What this should allow you to do, but I'm not exactly familiar with how this behaves over a shared object boundary, is you can replace the stream buffer of cerr with your own.

::std::istringstream log_stream;
auto old_rdbuf = ::std::cerr.rdbuf(log_stream.rdbuf());
// Test code...
::std::getline(log_stream, log_line);
// etc...
::std::cerr.rdbuf(old_rdbuf);

Reply:
@mreddington I have tried your approach (check last commit) but it does not work fine. There are no lines when reading from the `istringstream`, it seems empty. Am I missing something?

Reply:
@alb3rtobr I'm probably missing something. Streams are mostly brilliant until you start mucking with their details. I'm glad you tried it. I think I'll figure this one out and then make a separate ticket for it. We have a lot of testing that would benefit from internal log redirection. Thanks, mate.

Reply:
@mreddington No problem, thanks for the suggestion anyway. Im going to revert the last commit then.

Reply:
@mreddington Please remember to approve the PR if its fine for you :)